Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt

```diff
@@ -1,17 +1,48 @@
 package de.rki.coronawarnapp.ccl.dccwalletinfo.update
 
-import de.rki.coronawarnapp.ccl.dccwalletinfo.update.DccWalletInfoUpdateTask.DccWalletInfoUpdateTriggerType.TriggeredAfterCertificateChange
+import de.rki.coronawarnapp.ccl.dccwalletinfo.DccWalletInfoCleaner
+import de.rki.coronawarnapp.ccl.dccwalletinfo.calculation.DccWalletInfoCalculationManager
 import de.rki.coronawarnapp.ccl.dccwalletinfo.update.DccWalletInfoUpdateTask.DccWalletInfoUpdateTriggerType.TriggeredAfterConfigUpdate
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificates
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificatesProvider
 import de.rki.coronawarnapp.tag
 import de.rki.coronawarnapp.task.TaskController
 import de.rki.coronawarnapp.task.common.DefaultTaskRequest
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.launch
 import timber.log.Timber
 import javax.inject.Inject
+import javax.inject.Singleton
 
+@Singleton
 class DccWalletInfoUpdateTrigger @Inject constructor(
-    private val taskController: TaskController
+    personCertificateProvider: PersonCertificatesProvider,
+    @AppScope appScope: CoroutineScope,
+    private val taskController: TaskController,
+    private val dccWalletInfoCalculationManager: DccWalletInfoCalculationManager,
+    private val dccWalletInfoCleaner: DccWalletInfoCleaner
 ) {
 
+    init {
+        appScope.launch {
+            personCertificateProvider.personCertificates
+                .drop(1) // Drop first value on App start that causes unnecessary calculation
+                .distinctUntilChangedBy { it.sortedQrCodeHashSet }
```

Comment:
Maybe a short comment about which certificate changes will lead to a recalculation would be helpful here.